test/e2e/test/elasticsearch/checks_es.go

```diff
@@ -211,3 +236,84 @@ func compareMemoryLimit(topologyElement esv1.NodeSet, cgroupMemoryLimitsInBytes
 
 	return expectedBytes == cgroupMemoryLimitsInBytes
 }
+
+// compareCPULimit compares the CPU limit specified in a nodeSet with the limit set in the CPU control group at the OS level
+func compareCPULimit(topologyElement esv1.NodeSet, cgroupCPU struct {
+	CFSPeriodMicros int `json:"cfs_period_micros"`
+	CFSQuotaMicros  int `json:"cfs_quota_micros"`
+}) bool {
+	var expectedCPULimit *resource.Quantity
+	for _, c := range topologyElement.PodTemplate.Spec.Containers {
+		if c.Name == esv1.ElasticsearchContainerName {
+			expectedCPULimit = c.Resources.Limits.Cpu()
+		}
+	}
+	if expectedCPULimit == nil {
+		// no expected cpu, consider it's ok
+		return true
+	}
+
+	cgroupCPULimit := float64(cgroupCPU.CFSQuotaMicros) / float64(cgroupCPU.CFSPeriodMicros)
+	fmt.Println("{\"compareCPULimit\": \"", expectedCPULimit.AsApproximateFloat64() == cgroupCPULimit, "\"}")
+	fmt.Println("{\"compareCPULimit\": \"", expectedCPULimit.AsApproximateFloat64(), "\"}")
+	fmt.Println("{\"compareCPULimit\": \"", cgroupCPULimit, "\"}")
+	return expectedCPULimit.AsApproximateFloat64() == cgroupCPULimit
+}
+
+// compareResources compares the resources specified in the Elasticsearch resource with the resources
+// specified in the pods
+func compareResources(topologyElement esv1.NodeSet, pods []corev1.Pod) bool {
+	var expectedResources *corev1.ResourceRequirements
+	for _, c := range topologyElement.PodTemplate.Spec.Containers {
+		container := c
+		if c.Name == esv1.ElasticsearchContainerName {
+			expectedResources = &container.Resources
+		}
+	}
+	if expectedResources == nil {
+		expectedResources = &nodespec.DefaultResources
+	}
+	for _, pod := range pods {
+		for _, c := range pod.Spec.Containers {
+			if c.Name == esv1.ElasticsearchContainerName {
+				fmt.Println("{\"compareResources\": \"", reflect.DeepEqual(*expectedResources, c.Resources), "\"}")
+				fmt.Println("{\"compareResources\": \"", *expectedResources, "\"}")
+				fmt.Println("{\"compareResources\": \"", c.Resources, "\"}")
+				if !reflect.DeepEqual(*expectedResources, c.Resources) {
```

Comment:
It is not possible to compare 2 quantities using `DeepEqual` since a same value may have different internal representations. For example, '1000m' and '1' are semantically equal, but their representations are different.

```suggestion
				if !equality.Semantic.DeepEqual(*expectedResources, c.Resources) {
```